File: backend/models/position.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class PositionRecord:
    beacon_mac: str
    person_id: str
    person_type: str          # "guard" | "worker" | "forklift"
    x: float
    y: float
    zone: str
    timestamp: str            # ISO 8601
    predicted_x: Optional[float] = None
    predicted_y: Optional[float] = None
    pixel_x: Optional[float] = None   # metres × scale_pixels_per_meter from active floor plan
    pixel_y: Optional[float] = None
    is_stationary: bool = False
    floor_id: str = ""
    building_id: str = ""
    label: str = ""
    radius_m: Optional[float] = None   # set when position is a single-AP proximity estimate
    is_approximate: bool = False       # True when x/y is the anchor AP, not a multilateration solve
    anchor_ap_mac: Optional[str] = None  # MAC of the strongest AP this proximity estimate is anchored to (None for exact solves)

File: frontend/types/position.ts
export interface PositionRecord {
  beacon_mac: string;
  person_id: string;
  person_type: "guard" | "worker" | "forklift";
  x: number;
  y: number;
  zone: string;
  timestamp: string;         // ISO 8601
  predicted_x: number | null;
  predicted_y: number | null;
  pixel_x: number | null;   // metres × scale_pixels_per_meter; null until floor plan is calibrated
  pixel_y: number | null;
  is_stationary: boolean;
  floor_id: string;
  building_id: string;
  label: string;
  radius_m: number | null;
  is_approximate: boolean;
}
